pyClient/Makefile

```diff
@@ -4,6 +4,9 @@
 setup:
 	pip install --upgrade pip
 	pip install -e .
+	python -c "from solcx import install_solc_pragma; \
```

Comment:
Turns out that we have the following error in Travis:

`github ''API rate limit exceeded for 35.196.72.151. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)'`

I checked and 35.196.72.151 is a travis IP. The only way to remove this problem, that sporadically arises, is to avoid that these instructions are executed in the test. Any clue?

Reply:
Did you find a way to address this?
Let's make a `check_and_install_solc` function in contracts to check the installed version and only call install if necessary, to avoid this query.  Also, I think it makes sense to call that `check_and_install_solc` function from within the compile functions, so that we don't do this install unless it's genuinely required.

Reply:
It seems that `py-solcx` already checks if the compiler is installed in the system, so it does it only if it's required.  A solution could be to generate a API token for the script ( https://github.blog/2013-05-16-personal-api-tokens/) and add it to the repo. Just realized it was suggested in the pysolcx codebase: https://github.com/iamdefinitelyahuman/py-solc-x/blob/32dfb7b8c378b61d143d44e1bf49da25c42fd151/solcx/install.py#L164

Reply:
The github api call has been removed, so we should be fine now. Travis run without problems after the last commit. I'll close the conversion.